makefile.shade

```diff
@@ -1348,39 +1327,96 @@ functions @{
         return exitCode;
     }
 
-    HashSet<string> ListDependencyAssemblies(string projectFolder, string framework, string coreclrFolder = null)
+    HashSet<string> ListDependencyAssemblies(string projectFolder, RuntimeTarget target, string ROOT)
     {
-        var tmpFile = Path.GetTempFileName();
-        var argument = string.Format("list {0} --framework {1} --assemblies", projectFolder, framework);
-
-        if (coreclrFolder != null)
+        var runtimeTarget = "";
+        var args = "--runtime {0}";
```

Comment:
It seems confusing to have this here. Can you merge the declaration with the full initialization down on line 1348?